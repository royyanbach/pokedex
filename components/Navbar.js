import React, { Component } from "react";

export default class Navbar extends Component {
  constructor(props) {
    super(props);
  }

  render() {
    return (
      <nav className="navbar">
        <div className="container">
          <div className="navbar__content">
            <h1 className="hero">Pokedex</h1>
            { this.props.children }
          </div>
        </div>
      </nav>
    )
  }
}
