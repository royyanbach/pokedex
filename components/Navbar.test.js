import React from 'react';
import renderer from 'react-test-renderer';

import Navbar from './Navbar';

it('Render navbar correctly', () => {
  const totalDisplayedPokemon = 100;
  const totalPokemon = 1000;

  const component = renderer.create(
    <Navbar totalDisplayedPokemon={ totalDisplayedPokemon } totalPokemon={ totalPokemon }>Hello world</Navbar>,
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
