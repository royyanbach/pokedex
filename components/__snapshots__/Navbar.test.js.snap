// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render navbar correctly 1`] = `
<nav
  className="navbar"
>
  <div
    className="container"
  >
    <div
      className="navbar__content"
    >
      <div>
        <h1
          className="hero"
        >
          Pokedex
        </h1>
        <p
          className="total-information"
        >
          Showing 
          <strong>
            100
          </strong>
           from total of 
          <strong>
            1000
          </strong>
           pokemons
        </p>
      </div>
      Hello world
    </div>
  </div>
</nav>
`;
